model/task_queue.go

```diff
@@ -121,6 +123,26 @@ func shouldRunTaskGroup(taskId string, spec TaskSpec) bool {
 	return false
 }
 
+func (self *TaskQueue) BlockTaskGroupTasks(spec TaskSpec, taskID string) error {
+	catcher := grip.NewBasicCatcher()
+	for _, it := range self.Queue {
+		if spec.Group == it.Group && spec.BuildVariant == it.BuildVariant && spec.Version == it.Version {
+			catcher.Add(errors.Wrapf(self.DequeueTask(it.Id), "problem dequeueing task %s", it.Id))
+			t, err := task.FindOneId(it.Id)
+			if err != nil {
+				catcher.Add(errors.Wrapf(err, "problem finding task %s", it.Id))
+				continue
+			}
+			if t == nil {
+				catcher.Add(errors.Errorf("got nil task for %s", it.Id))
+				continue
+			}
+			catcher.Add(t.AddDependency(task.Dependency{TaskId: taskID, Status: evergreen.TaskSucceeded}))
```

Comment:
Should we add the dependency first to prevent the scheduler from enqueuing again?